### DISCUSSION TITLE
Roadmap of upcoming features

### DISCUSSION THREAD

**Discussion by Speierers** [mitsuba-renderer/mitsuba3#121] [General]
Hi everyone,

In this post we would like to share a roadmap of features we are hoping to add in the next release.

This list will grow over time as we make progress on the developement.

- Make Mitsuba extension modules easy to install out-of-band.
  --> e.g. `pip install mitsuba.fabric` to add a hypothetical fabric BSDF plugin to your installation
- Port the binding generation from `pybind11` to `nanobind` (both in Dr.Jit and Mitsuba 3).
- Add a few more complex production scenes.
- .. to be continued ..

**Comment by xelatihy** [comment]
My two cents:

1. I am in the process of converting some large scenes to mitsuba. If there is interest, we should touch base.
2. We have tested both the Optix and Intel denser, and at least at the time of testing the latter looked better (unscientifically).

**Reply by dvicini** [reply]
Hi Fabio,
Having large scenes sounds interesting. Let us know if you encounter any issues. If you end up hosting a collection of interesting scenes, we are (I assume) happy to include a link to it somewhere in the documentation.

As far as the denoiser is concerned: having the intel denoiser would be great, it's always better to have features work across different platforms. However, last time I checked, setting up the intel denoiser was rather tedious (e.g., it uses ISPC). It seems like it would be easier for someone to create a prebuilt Python package of it that people could then install separately, rather than include it into Mitsuba. The advantage of the OptiX denoiser is that using it is a simple as adding appropriate Nvidia driver API calls.

**Comment by stefanjp** [comment]
Is integration with pytorch on the roadmap? I have seen a git feature branch on the topic when mitsuba 3 has been released.
Edit: The branch is still publicly accessible and resides in the DrJit repository (from_to_torch).

**Reply by Speierers** [reply]
Yes this is on the roadmap as well actually!

**Comment by errissa** [comment]
Any idea of if/when Mitsuba will move to Embree 4? And are there any plans to support a SYCL backend so as to take advantage of Intel GPUs?

**Reply by njroussel** [reply]
Hi @errissa 

There is no plan to update our Embree version. We typically only upgrade if there is some important new feature or bugfix which we need. AFAIK, there isn't anything of interest for us in Embree 4 (yet).

As for backends, we are still planning on adding a Metal backend first. There aren't any plans for other backends after that unfortunately.

**Reply by errissa** [reply]
Thanks for the info. A Metal backend is very exciting!

**Comment by njroussel** [comment]
Unpinned this discussion for now - it's outdated. Will pin it again once we decide on an updated list.
